dev-lang/ispc/ispc-1.14.1.ebuild

```diff
@@ -22,10 +24,8 @@ LICENSE="BSD BSD-2 UoI-NCSA"
 SLOT="0"
 IUSE="examples"
 
-RDEPEND="
-	>=sys-devel/clang-3.0:*
-	>=sys-devel/llvm-3.0:*
-	"
+RDEPEND="<sys-devel/clang-11:=[llvm_targets_AMDGPU(+)]"
```

Comment:
Why do you pass the llvm_target? I successfully compiled it against clang/llvm without this target defined.

Reply:
With the current settings ispc needs AVX2 on x86_64 or x86. I thought it was coming from that target, but now I see that it probably should be the x86 target instead, which includes both x86_64 and x86.

Reply:
AVX2 seems not strictly needed. I'm having a bulldozer family CPU which doesn't support AVX2 (according to cpuid2cpuflags and /proc/cpuinfo) and the package builds fine.

Reply:
Thanks, I think that means we can just go with llvm/clang 10 without the target specified.

Maybe if you don't have AVX2 it automatically builds with SSE4 or something else instead?

Perhaps this is for down the track, but I would be interested to know if you can run open image denoiser with blender on your cpu, or whether it is necessary to add USE flags to control which version is built for either ispc or oidn in order to do so. The install_deps script in blender builds oidn with AVX2 support by default, but I don't know if it a hard requirement or not.

Reply:
It builds for AVX support.

Currently I'm not able to build oidn from your overlay. The denoiser works however with the official binary blender package. The issues are the same, like when I was trying to build the ispc examples and tests and point to an internal ispc compiler error. I will probably report this upstream.

```FAILED: local__core/output_copy.dev.o local__core/output_copy.dev_sse4.o local__core/output_copy.dev_avx2.o local__core/output_copy.dev_avx512skx.o output_copy_ispc.h 
cd /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4_build && /usr/bin/cmake -E make_directory /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4_build/local__core && /usr/bin/ispc -I /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4 --arch=x86-64 --addressing=64 -O3 --target=sse4-i32x8,avx2-i32x8,avx512skx-i32x16 --woff --opt=fast-math --pic --opt=disable-assertions -h /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4_build/output_copy_ispc.h -MMM /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4_build/local__core/output_copy.dev.idep -o /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4_build/local__core/output_copy.dev.o /var/tmp/portage/media-libs/oidn-1.2.4/work/oidn-1.2.4/core/output_copy.ispc
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
/var/tmp/portage/dev-lang/ispc-1.14.1/work/ispc-1.14.1/src/main.cpp(372): FATAL ERROR: Unhandled signal sent to process; terminating.
***
*** Please file a bug report at https://github.com/ispc/ispc/issues
*** (Including as much information as you can about how to reproduce this error).
*** You have apparently encountered a bug in the compiler that we'd like to fix!
***
```

This exception throws whenever an ispc program is compiled (ispc examples, tests, or oidn programs using ispc).

I noticed this, because cuda isn't working in blender, neither with the build from your overlay, nor with the offical binary. But I hope I found the issue. Current cuda-11.1 needs >=nvidia-driver-450.80.02, but we only have 455.38-r1. Now trying to downgrade nvidia-cuda-toolkit to 11.0.

Reply:
I tried the same commands you pasted above, and it completes without error on my machine. The local_core directory is filled with
```
output_copy.dev.idep  output_copy.dev_avx2.o       output_copy.dev_sse4.o
output_copy.dev.o     output_copy.dev_avx512skx.o
```

I notice there is avx2 in your target, do you think this might be a contributing factor?

I saw a warning the other day that blender only officially supports nvidia-cuda-toolkit 10 and 10.1, see https://developer.blender.org/D8063. If downgrading to 10 fixes your issue we might need to patch the blender ebuild. Having said that, it seems to work with 11.1 and nvidia-drivers-455.28-r1 for me.

There is also the issue with the gpu architectures supported which changed with CUDA 11, see https://bugs.gentoo.org/744517

Reply:
> I notice there is avx2 in your target, do you think this might be a contributing factor?

Might be. I noticed this file too in the error output and was thinking whether this might be an issue.

> I saw a warning the other day that blender only officially supports nvidia-cuda-toolkit 10 and 10.1, see https://developer.blender.org/D8063. If downgrading to 10 fixes your issue we might need to patch the blender ebuild. Having said that, it seems to work with 11.1 and nvidia-drivers-455.28-r1 for me.
> 
> There is also the issue with the gpu architectures supported which changed with CUDA 11, see https://bugs.gentoo.org/744517

Yep they changed a lot, but my GPU should still be compatible. It's a compute platform 35, Kepler architecture, GK208 type, GeForce GT 730, which should still be supported according to https://en.wikipedia.org/wiki/CUDA#GPUs_supported

Reply:
I have fixed the unnecessary dependency on llvm targets in both ebuilds.

I am not sure that the patch to restrict llvm find_package is necessary. I have tested compiling ispc without it on a few systems now and it never chooses LLVM 11 for me.